eth2util/network.go

```diff
@@ -63,8 +64,14 @@ var supportedNetworks = []Network{
 	Mainnet, Goerli, Gnosis, Sepolia, Holesky,
 }
 
+// AddTestNetwork adds given network to list of supported networks.
+// Note: This function should only be called once.
```

Comment:
Why?

Reply:
Because in this function we are writing to `supportedNetworks` slice. This can introduce race condition if called twice at the same time by different goroutines as it is not protected by and mutex/once.
Other option can be to have `sync.Once` so that this _will_ be called once.
Open to suggestions to improve godoc for more clarity.

The usage of AddTestNetwork is to add the testnet config as the first step for any process (create cluster OR charon run) so that other functions in this file will behave as expected.

Reply:
My first suggestion is to fix the go doc: this function can be called more than once, but it can’t be called *concurrently*. 

One thing that we could do is “seal” the slice: add a bool flag that is true when it’s sealed. If true, don’t allow appends. 

Another solution (arguably simpler to reason about) is to add locking around slice access. 

Honestly even just documenting the racy nature of this function would suffice, but adding a lock and remove the ambiguity is the most complete solution IMO.

Reply:
yeah, I initially went with sync.Once but then thought that just documenting would suffice. I'll go with fixing the godoc.

Reply:
yeah that is why I went with `sync.Once` to protect this function. Now i am thinking about an integration test which can call `app.Run(..)` in different goroutines 🤔 

Cool, I am going with adding a lock 👍

Reply:
> Now i am thinking about an integration test which can call app.Run(..) in different goroutines 🤔

Now this is interesting 👀 

The race detector would go wild!

Reply:
Done, I added lock 👍